### PR TITLE
Highlight that julia integers are machine integers

### PR DESCRIPTION
**Pull request by oxinabox**
There have been 2 recent discourse threads about this 
https://discourse.julialang.org/t/weird-result-in-power-operator/20824
https://discourse.julialang.org/t/julia-messes-up-integer-exponents/20773/68

### CODE REVIEW

**Review by StefanKarpinski — APPROVED** [review]

File: doc/src/manual/noteworthy-differences.md [hunk 1]
```diff
@@ -21,6 +21,9 @@ may trip up Julia users accustomed to MATLAB:
     point numbers. Arbitrarily large integer literals are supported. As a result, some operations
     such as `2^-1` will throw a domain error as the result is not an integer (see [the FAQ entry on domain errors](@ref faq-domain-errors)
     for details).
+  * Integer literals are of type `Int` which corresponds to the machine integer type (`Int32` or `Int64`).
+    This means it will overflow, such that `2^64 == 0`. If you need larger values use another appropriate type,
```
Comment (KristofferC) [inline doc/src/manual/noteworthy-differences.md]:
```suggestion
    This means that they will overflow, such that `2^64 == 0`. If you need larger values use another appropriate type,
```

to be consistent with the subject from the previous sentence "Integer literals".

File: doc/src/manual/noteworthy-differences.md [hunk 2]
```diff
@@ -21,6 +21,9 @@ may trip up Julia users accustomed to MATLAB:
...
+    such as [Int128](@ref), `BigInt`(@ref) or a floating point value like [Float64](@ref).
```
Comment (KristofferC) [inline doc/src/manual/noteworthy-differences.md]:
```suggestion
    such as [`Int128`](@ref), [`BigInt`](@ref) or a floating point value like [`Float64`](@ref).
```

Comment (KristofferC) [reply doc/src/manual/noteworthy-differences.md]: Or maybe not all of these have refs?

File: doc/src/manual/noteworthy-differences.md (hunk 1 above)
Comment (oxinabox) [reply doc/src/manual/noteworthy-differences.md]: It is actually wrong, so I have rewritten  it.

```
julia> 10000000000000000000000000000000000000000 |> typeof
BigInt

julia> 100000000000000000000000000000 |> typeof
Int128

julia> 100000000 |> typeof
Int64
```

File: doc/src/manual/noteworthy-differences.md (hunk 2 above)
Comment (oxinabox) [reply doc/src/manual/noteworthy-differences.md]: correct

File: doc/src/manual/noteworthy-differences.md [hunk 3]
```diff
@@ -220,6 +220,9 @@ For users coming to Julia from R, these are some noteworthy differences:
     On the other hand, the function `g(x=[1,2]) = push!(x,3)` returns `[1,2,3]` every time it is called
     as `g()`.
   * In Julia `%` is the remainder operator, whereas in Python it is the modulus.
+  * The commonly used `Int` type which corresponds to the machine integer type (`Int32` or `Int64`).
```
Comment (JeffBezanson) [inline doc/src/manual/noteworthy-differences.md]:
```suggestion
  * The commonly used `Int` type corresponds to the machine integer type (`Int32` or `Int64`).
```

File: doc/src/manual/noteworthy-differences.md [hunk 4]
```diff
@@ -220,6 +220,9 @@ For users coming to Julia from R, these are some noteworthy differences:
...
+    This means it will overflow, such that `2^64 == 0`. If you need larger values use another appropriate type,
+    such as `Int128`, [`BigInt`](@ref) or a floating point value like `Float64`.
```
Comment (JeffBezanson) [inline doc/src/manual/noteworthy-differences.md]:
```suggestion
    such as `Int128`, [`BigInt`](@ref) or a floating point type like `Float64`.
```

**Review by JeffBezanson — APPROVED** [review]

**Pull request merged**